jax/numpy/lax_numpy.py

```diff
@@ -2573,6 +2573,10 @@ def diag_indices(n, ndim=2):
                      .format(ndim))
   return (lax.iota(int_, n),) * ndim
 
+@_wraps(np.diag_indices_from)
+def diag_indices_from(arr):
+  return diag_indices(arr.shape[0], ndim=arr.ndim)
```

Comment:
numpy returns an error if the array is not square:
```python
>>> import numpy as np
>>> np.diag_indices_from(jnp.zeros((5, 8)))
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-5-1c7a827c990f> in <module>()
      1 import numpy as np
----> 2 np.diag_indices_from(jnp.zeros((5, 8)))

<__array_function__ internals> in diag_indices_from(*args, **kwargs)

/usr/local/lib/python3.6/dist-packages/numpy/lib/index_tricks.py in diag_indices_from(arr)
    980     # all dimensions equal, so we check first.
    981     if not alltrue(diff(arr.shape) == 0):
--> 982         raise ValueError("All dimensions of input must be of equal length")
    983 
    984     return diag_indices(arr.shape[0], arr.ndim)

ValueError: All dimensions of input must be of equal length
```
We should have a similar check here.